src/swap/v2/router/swap-router.ts

```diff
@@ -0,0 +1,203 @@
+import algosdk, {ALGORAND_MIN_TX_FEE, getApplicationAddress, Transaction} from "algosdk";
+import AlgodClient from "algosdk/dist/types/src/client/v2/algod/algod";
+
+import {CONTRACT_VERSION} from "../../../contract/constants";
+import {ALGO_ASSET_ID} from "../../../util/asset/assetConstants";
+import {getAssetId, isAlgo} from "../../../util/asset/assetUtils";
+import {SupportedNetwork} from "../../../util/commonTypes";
+import {getValidatorAppID} from "../../../validator";
+import {SwapType} from "../../constants";
+import {
+  FetchSwapRouteQuotesPayload,
+  FetchSwapRouteQuotesResponse,
+  GenerateSwapRouterTxnsParams
+} from "../../types";
+import {
+  V2_SWAP_APP_CALL_ARG_ENCODED,
+  V2_SWAP_APP_CALL_SWAP_TYPE_ARGS_ENCODED,
+  V2_SWAP_ROUTER_APP_ARGS_ENCODED
+} from "../constants";
+import {SWAP_ROUTER_INNER_TXN_COUNT, TINYMAN_ANALYTICS_API_BASE_URLS} from "./constants";
+import {
+  getAssetInFromSwapRoute,
+  getAssetOutFromSwapRoute,
+  getSwapRouterAppID
+} from "./util";
+
+export async function generateSwapRouterAssetOptInTransaction({
```

Comment:
I think this function requires a short description and the name may be a little misleading.

```
Generates txns that would opt in the Swap Router Application to the assets used in the swap router
```